chainercv/datasets/directory_parsing_classification_dataset.py

```diff
@@ -0,0 +1,134 @@
+import numpy as np
+import os
+
+import chainer
+from chainercv.utils import read_image
+
+
+def parse_label_names(root, numerical_sort=False):
+    """Get label names from directories that are named by them.
+
+    The label names are names of the directories that locate a layer below the
+    root.
+
+    The label names can be used together with
+    :class:`chainercv.datasets.DirectoryParsingClassificationDataset`.
+    An index of the label names correspond to a label id
+    obtained from the dataset.
+
+    Args:
+        root (str): The root directory.
+        numerical_sort (bool): Label names are sorted numerically.
+            This means that :obj:`'2'` is before :obj:`10`,
+            which is not the case when string sort is used.
+            The default value is :obj:`False`.
+
+    Retruns:
+        list of strings:
+        Sorted names of classes.
+
+    """
+    label_names = [d for d in os.listdir(root)
+                   if os.path.isdir(os.path.join(root, d))]
+
+    if not numerical_sort:
+        label_names.sort()
+    else:
+        label_names = [int(name) for name in label_names]
+        label_names.sort()
+        label_names = [str(name) for name in label_names]
+    return label_names
+
+
+def _ends_with_img_ext(filename):
+    img_extensions = ['.jpg', '.jpeg', '.png', '.ppm', '.bmp']
+    return any(os.path.splitext(filename)[1].lower().endswith(extension) for
+               extension in img_extensions)
+
+
+def _parse_classification_dataset(root, label_names,
+                                  check_img_file=_ends_with_img_ext):
+    # Use label_name_to_idx for performance.
+    label_name_to_idx = {label_names[i]: i for i in range(len(label_names))}
+
+    img_paths = []
+    labels = []
+    for label_name in os.listdir(root):
```

Comment:
Why don't you use `for label, label_name in enumerate(label_names)`? If we use this, we can remove `label_name_to_idx`.